### DISCUSSION TITLE
Default to using software gamma control when DDC information missing

### DISCUSSION THREAD

**Discussion by benjamingwynn** [MonitorControl/MonitorControl#1670] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues

### Is your feature request related to a problem? Please describe

I use a screen without proper EDID/DDC information/control, the second display here:

![Screenshot 2024-11-10 at 11 39 17](https://github.com/user-attachments/assets/5bba4a83-b354-447a-84a0-759a406e006c)

Sometimes MonitorControl seems to forget which monitor this is, probably because it lacks proper identifiying information.

### Describe the solution you'd like

It'd be good if MonitorControl were to just default to software control for monitors with invalid EDID/DDC information, or if MonitorControl were able to store the fact this monitor requires software control consistently.

### Describe alternatives you've considered

I have no tried alteratives to MonitorControl, I haven't done any detailed testing either. If there are logs I can grab or futher steps to run through, please let me know!

### Anything else?

Thank you for all the work bringing third-party monitors up to native-feeling support for macOS!

**Comment by waydabber** [comment]
The app falls back to gamma control if it thinks DDC is not available. For more nuanced display identification methods you might want to try [BetterDisplay](https://github.com/waydabber/BetterDisplay#readme) where you can configure various identification methods for the display (under Settings/Displays/display/Advanced). Choose one that works reliably with the display.

**Reply by benjamingwynn** [reply]
Hi @waydabber, thanks for getting back to me so quickly. DDC is not available on this display, I'm using [this adapter](https://www.startech.com/en-gb/display-video-adapters/cdp2dvidp) for use with a 1440p display over dual link DVI.

Should MonitorControl be properly detecting that DDC isn't available for this display?

**Reply by benjamingwynn** [reply]
I'm not super familiar with the code base but perhaps having [isDummy](https://github.com/MonitorControl/MonitorControl/blob/71b8c5ae51955d17b05688371ea7b899b684bdbd/MonitorControl/Support/DisplayManager.swift#L402) return true for an empty string could work?

**Reply by waydabber** [reply]
Yes, you can simply fork the code and add a custom line not to enable DDC if the display's name is an empty string. You can easily compile the version with XCode to run locally.
